lms/services/blackboard_api/_schemas.py

```diff
@@ -18,11 +18,15 @@ class Meta:
             unknown = EXCLUDE
 
         id = fields.Str(required=True)
-        display_name = fields.Str(data_key="name", required=True)
-        updated_at = fields.Str(data_key="modified", required=True)
+        name = fields.Str(required=True)
+        modified = fields.Str(required=True)
+        downloadUrl = fields.Str(required=True)
 
     results = fields.List(fields.Nested(FileSchema), required=True)
 
     @post_load
     def make_object(self, data, **_kwargs):  # pylint:disable=no-self-use
+        for file_ in data["results"]:
+            file_["id"] = f"blackboard://content-resource/{file_['id']}/"
```

Comment:
This is where we turn Blackboard file IDs into `blackboard://...` URLs before returning the list of them to the frontend when a teacher is creating an assignment.

They are just opaque strings to the frontend, it knows nothing about the `blackboard://...` format.

When the teacher selects a file the frontend submits the `blackboard://...` URL to the `configure_module_item()` view as the `document_url` and it gets saved in the DB. It's an opaque string to the `configure_module_item()` view and to the DB as well.

One weird thing about this is that `blackboard://...` is the file `id` in the list files response from our proxy API but it's the `document_url` in the `ModuleItemConfiguration` table :man_shrugging: 

The nice thing about doing this in the schema is that both the `list_files()` and `public_url()` methods use this schema. The `public_url()` method receives a `blackboard://...` URL as the `file_id`, gets the list of files from Blackboard (passing them through the schema so their IDs get turned into `blackboard://...` URLs), and compares the `blackboard://...` IDs of the list of files from the schema to the `blackboard://...` ID that it was given in its `file_id` argument. This means:

1. We only every format `blackboard://...` URLs. We never have to parse them
2. This one line in the schema is the only place in the code that knows about the `blackboard://...` format

The only exception to (2) is that `JSConfig` has to look for strings that start with `blackboard://` to know when a Blackboard Files assignment is being launched:

https://github.com/hypothesis/lms/blob/1a0324d8e0e8213bb0f27c46a2561e20206a17af/lms/resources/_js_config.py#L65-L73

These are the only two places in the code where the `"blackboard://"` string exists